x-pack/legacy/plugins/lens/public/indexpattern_plugin/operations/definitions/date_histogram.test.tsx

```diff
@@ -14,19 +14,26 @@ import { IStorageWrapper } from 'src/plugins/kibana_utils/public';
 import { createMockedIndexPattern } from '../../mocks';
 import { IndexPatternPrivateState } from '../../types';
 
-jest.mock('ui/new_platform', () => ({
-  npStart: {
-    core: {
-      uiSettings: {
-        get: (path: string) => {
-          if (path === 'histogram:maxBars') {
-            return 10;
-          }
+jest.mock('ui/new_platform', () => {
+  const { coreMock } = require('src/core/public/mocks'); // eslint-disable-line @typescript-eslint/no-var-requires
```

Comment:
Can you explain what this is doing? I've only looked at this file so far.

Reply:
Yep, so unfortunately it's a very long answer to a very simple question 😄 

For context, here's what the text failure looked like:
```
 FAIL  legacy/plugins/lens/public/indexpattern_plugin/operations/definitions/date_histogram.test.tsx
  ● Test suite failed to run

    TypeError: Cannot read property 'core' of undefined

      21 | import { ToastNotifications } from './toast_notifications';
      22 | 
    > 23 | export const toastNotifications = new ToastNotifications(npSetup.core.notifications.toasts);
         |                                                                  ^
      24 | 

      at Object.<anonymous> (../src/legacy/ui/public/notify/toasts/toasts.ts:23:66)
      at Object.<anonymous> (../src/legacy/ui/public/notify/toasts/index.ts:20:1)
      at Object.<anonymous> (../src/legacy/core_plugins/data/public/search/fetch/handle_response.tsx:23:1)
      at Object.<anonymous> (../src/legacy/core_plugins/data/public/search/fetch/call_client.ts:22:1)
      at Object.<anonymous> (../src/legacy/core_plugins/data/public/search/fetch/fetch_soon.ts:20:1)
      at Object.<anonymous> (../src/legacy/core_plugins/data/public/search/fetch/index.ts:20:1)
      at Object.<anonymous> (../src/legacy/core_plugins/data/public/search/search_source/search_source.ts:77:1)
      at Object.<anonymous> (../src/legacy/core_plugins/data/public/search/search_source/index.ts:20:1)
      at Object.<anonymous> (../src/legacy/core_plugins/data/public/search/search_service.ts:21:1)
      at Object.<anonymous> (../src/legacy/core_plugins/data/public/search/index.ts:21:1)
      at Object.<anonymous> (../src/legacy/core_plugins/data/public/plugin.ts:21:1)
      at Object.<anonymous> (../src/legacy/core_plugins/data/public/index.ts:21:1)
      at Object.<anonymous> (../src/legacy/ui/public/agg_types/utils.ts:20:1)
      at Object.<anonymous> (legacy/plugins/lens/public/indexpattern_plugin/operations/definitions/date_histogram.tsx:12:1)
      at Object.<anonymous> (legacy/plugins/lens/public/indexpattern_plugin/operations/definitions/index.ts:12:1)
      at Object.<anonymous> (legacy/plugins/lens/public/indexpattern_plugin/operations/definitions/date_histogram.test.tsx:9:1)
```

What's happening is:
- `ui/courier` files moved into `core_plugins/data/public`
- the `ui/agg_types` util `isValidInterval`, which `date_histogram` depends on, imports a file from `core_plugins/data` (this is completely unrelated to courier)
- BUT now that courier has moved into the shim plugin, any deeply nested dependencies inside of search source which rely on core services are actually resolved when someone does a static import from `core_plugins/data/public`, like `ui/agg_types` is doing.
- what this means is that any legacy plugins who are relying on anything which relies on the new platform will need to mock new platform services during the migration process (basically everyone at some point)
- this is why `jest.mock('ui/new_platform')` exists
- because lens `date_histogram` is using agg types, it needs the `ui/new_platform` mock in the test
- but one of the assertions specifically relies on the mocked return value of `10` for the uiSettings `histogram:maxBars`, so as a result there is an "incomplete" `ui/new_platform` mock for the file
- so we need to merge the existing mock with other mocked `ui/new_platform` stuff to prevent tests from breaking 
- since babel hoists `jest`, you can't rely on variables in your mocks, hence the inline `require` for the core mock

I'm sure there's a more succinct/elegant way of handling the merging of these mocks, so please let me know if you have any thoughts!

Reply:
Thanks for the explanation, this makes sense to me. It sounds like we are testing too much here, but I appreciate that you've found a way to work around the test limitation. Some version of this comment could live in the code for future use, since we rarely use inline `require()` like you've added here.

Reply:
> Some version of this comment could live in the code for future use

Sounds good! I'll add a note. Really all of the uses of `jest.mock('ui/new_platform')` will need to go away during the migration process anyway, but I'll add a comment to be clearer on this.